File: sdk/typescript-guest/templates/greeting.ts
import type * as Contract from '../generated/interfaces/examples-greeting-api.js';
import { trim, utf8Length } from '../vendor/lsf/sdk/typescript-guest/runtime/text.js';

// lsf-example-begin: capsule
export const api: typeof Contract = {
  greet(name) {
    const clean = trim(name);
    if (!clean) throw 'Please enter a name.';
    if (utf8Length(clean) > 100) throw 'Use a name of at most 100 bytes.';
    return `Hello, ${clean}!`;
  },
};
// lsf-example-end: capsule

File: sdk/dotnet-guest/templates/greeting.cs
using ServiceWorld;
using Lsf.Guest;
namespace ServiceWorld.wit.Exports.examples.greeting;

// lsf-example-begin: capsule
public class ApiExportsImpl : IApiExports
{
    public static Result<string, string> Greet(string name)
    {
        string clean = Text.Trim(name);
        if (clean.Length == 0) return Result<string, string>.Err("Please enter a name.");
        if (Text.Utf8Length(clean) > 100) return Result<string, string>.Err("Use a name of at most 100 bytes.");
        return Result<string, string>.Ok($"Hello, {clean}!");
    }
}
// lsf-example-end: capsule
